code/game/machinery/computer/aifixer.dm

```diff
@@ -48,43 +47,43 @@
 
 		var/laws[0]
 		for(var/datum/ai_law/law in occupant.laws.all_laws())
-			laws.Add(list(list("law" = law.law, "number" = law.get_index())))
+			laws.Add(list("law" = law.law, "number" = law.get_index())))
 
 		data["laws"] = laws
 
 	return data
 
-/obj/machinery/computer/aifixer/Topic(href, href_list)
+/obj/machinery/computer/aifixer/tgui_act(action, params)
 	if(..())
-		return 1
-
-	if(href_list["fix"])
-		active = 1
-		while(occupant.health < 100)
-			occupant.adjustOxyLoss(-1, FALSE)
-			occupant.adjustFireLoss(-1, FALSE)
-			occupant.adjustToxLoss(-1, FALSE)
-			occupant.adjustBruteLoss(-1, FALSE)
-			occupant.updatehealth()
-			if(occupant.health >= 0 && occupant.stat == DEAD)
-				occupant.update_revive()
-				occupant.lying = 0
-				update_icon()
-			sleep(10)
-		active = 0
-		add_fingerprint(usr)
-
-	if(href_list["wireless"])
-		var/wireless = text2num(href_list["wireless"])
-		if(wireless == 0 || wireless == 1)
-			occupant.control_disabled = wireless
-
-	if(href_list["radio"])
-		var/radio = text2num(href_list["radio"])
-		if(radio == 0 || radio == 1)
-			occupant.aiRadio.disabledAi = radio
-
-	SSnanoui.update_uis(src)
+		return
+	switch(action)
+		if("fix")
+			active = 1
+			while(occupant.health < 100)
+				occupant.adjustOxyLoss(-1, FALSE)
+				occupant.adjustFireLoss(-1, FALSE)
+				occupant.adjustToxLoss(-1, FALSE)
+				occupant.adjustBruteLoss(-1, FALSE)
+				occupant.updatehealth()
+				if(occupant.health >= 0 && occupant.stat == DEAD)
+					occupant.update_revive()
+					occupant.lying = 0
+					update_icon()
+				sleep(10)
+			active = 0
+			add_fingerprint(usr)
+
+		if("wireless")
+			var/wireless = text2num("wireless")
+			if(wireless == 0 || wireless == 1)
```

Comment:
What are the possible values for this other than 0 or 1?

Reply:
I guess it was for href checks or something??? No clue, gonna just replace that with a
```
if(x)
    thing.to.change = thing```